lib/wda-mac.js

```diff
@@ -0,0 +1,265 @@
+import _ from 'lodash';
+import path from 'path';
+import { JWProxy, errors } from 'appium-base-driver';
+import { fs, logger, util, timing } from 'appium-support';
+import { SubProcess, exec } from 'teen_process';
+import { waitForCondition } from 'asyncbox';
+import { execSync } from 'child_process';
+
+const log = logger.getLogger('WebDriverAgentMac');
+
+const ROOT_DIR = path.basename(__dirname) === 'lib'
+  ? path.resolve(__dirname, process.env.NO_PRECOMPILE ? '..' : '../..')
+  : __dirname;
+const WDA_ROOT = path.resolve(ROOT_DIR, 'WebDriverAgentMac');
+const WDA_PROJECT_NAME = 'WebDriverAgentMac.xcodeproj';
+const WDA_PROJECT = (wdaRoot = WDA_ROOT) => path.resolve(wdaRoot, WDA_PROJECT_NAME);
+const SCRIPTS_ROOT = (wdaRoot = WDA_ROOT) => path.resolve(wdaRoot, 'Scripts');
+const BOOTSTRAP_SCRIPT = (wdaRoot = WDA_ROOT) => path.resolve(SCRIPTS_ROOT(wdaRoot), 'bootstrap.sh');
+const RUNNER_SCHEME = 'WebDriverAgentRunner';
+const DISABLE_STORE_ARG = 'COMPILER_INDEX_STORE_ENABLE=NO';
+const XCODEBUILD = 'xcodebuild';
+const CARTHAGE = 'carthage';
+const CARTHAGE_FRAMEWORKS_ROOT = (wdaRoot = WDA_ROOT) => path.resolve(wdaRoot, 'Carthage');
+const STARTUP_TIMEOUT_MS = 120000;
+const DEPS_LOAD_TIMEOUT_MS = 120000;
+const DEFAULT_SYSTEM_PORT = 10100;
+
+
+class WDAMacProxy extends JWProxy {
```

Comment:
this file all looks good, but if it duplicates a lot of what's in the xcui driver, it would be nice to find a way to address it, perhaps by putting the mac driver and the xcui driver in a monorepo where we can share code easily but still publish separate npm driver modules

Reply:
It was a deliberate decision. We already have xcuitest test driver, that depends heavily on the deprecated and non-maintained ios-driver. I would rather keep drivers isolated. Also, this in driver many things will be different from the mobile one. I tried to avoid using private interfaces and hacks as much as possible.